src/plugins/plan/Plan.vue

```diff
@@ -99,16 +99,23 @@ export default {
             this.setScaleAndPlotActivities();
         },
         resize() {
-            let clientWidth = this.$refs.plan.clientWidth;
-            if (this.options.clientWidth !== undefined) {
-                clientWidth = this.options.clientWidth;
-            }
-
+            let clientWidth = this.getClientWidth();
             if (clientWidth !== this.width) {
                 this.setDimensions();
                 this.updateViewBounds();
             }
         },
+        getClientWidth() {
+            let clientWidth = this.$refs.plan.clientWidth;
+
+            //this is a hack - need a better way to find the parent of this component
+            let parent = this.openmct.layout.$refs.browseObject.$el;
```

Comment:
I don't think this will work when the timeline is embedded within a display layout. It's going to calculate the size based on the wrong element.

Why do we need to get the size of the parent in JavaScript anyway? If we must use JS, could we just have a holder div in this component with width and height of 100% so that it will naturally expand to the size of its parent, and then get the dimensions from that?

Or, failing that, could we use `Element.closest()`? https://developer.mozilla.org/en-US/docs/Web/API/Element/closest Obviously creating a dependency on markup in a parent component is risky, but it might do for now until we can fix this properly.

Reply:
OK I'll go back to Element.closest(). Deep had suggested using the layout ref. :)